[1.1.x] Fix UBL compilation warnings

tcm0116 opened the pull request:
Fixes a few compilation warnings generated when compiling UBL with PlatformIO.

thinkyhead commented on Marlin/ultralcd.cpp:
```diff
@@ -2099,7 +2099,7 @@ void kill_screen(const char* lcd_msg) {
       ;
       sprintf_P(UBL_LCD_GCODE, PSTR("G26 C B%i H%i P"), temp, custom_hotend_temp);
-      lcd_enqueue_command_sram("G28");
+      lcd_enqueue_command_sram(PSTR("G28"));
```
Since `lcd_enqueue_command_sram` only works with strings that are stored in SRAM, this will not work. Try this instead:
```cpp
char * const g28 = "G28";
lcd_enqueue_command_sram(g28);
```

tcm0116 committed:
Fix UBL compilation warnings

tcm0116 commented:
@thinkyhead I added a `lcd_enqueue_commands_P` function to allow for enqueuing commands stored in program memory. To be consistent with `enqueue_and_echo_command`, I dropped the `_sram` from the end of the existing `lcd_enqueue_command` function.

thinkyhead commented on Marlin/Marlin_main.cpp:
```diff
@@ -848,9 +848,9 @@ static bool drain_injected_commands_P() {
+bool enqueue_and_echo_commands_P(const char * const pgcode) {
   injected_commands_P = pgcode;
-  drain_injected_commands_P(); // first command executed asap (when possible)
+  return drain_injected_commands_P(); // first command executed asap (when possible)
```
The status returned by `drain_injected_commands_P` only indicates that there are more commands in the passed string which may be queued in the next loop. It does not indicate success or failure.

thinkyhead commented on Marlin/ultralcd.cpp:
```diff
@@ -1683,12 +1683,18 @@ void kill_screen(const char* lcd_msg) {
 
+    void lcd_enqueue_commands_P(const char * const cmd) {
+      no_reentry = true;
+      while (enqueue_and_echo_commands_P(cmd)) idle();
```
Because of the way that `enqueue_and_echo_commands_P` is meant to be used, this could be very bad. To do what you're attempting, you must instead…
```cpp
enqueue_and_echo_commands_P(cmd);
while (drain_injected_commands_P()) idle();
```

thinkyhead commented on Marlin/ultralcd.cpp:
```diff
@@ -2099,8 +2105,8 @@ void kill_screen(const char* lcd_msg) {
       sprintf_P(UBL_LCD_GCODE, PSTR("G26 C B%i H%i P"), temp, custom_hotend_temp);
-      lcd_enqueue_command_sram("G28");
-      lcd_enqueue_command_sram(UBL_LCD_GCODE);
+      lcd_enqueue_commands_P(PSTR("G28"));
```
If this is the only cause for `lcd_enqueue_commands_P` to exist, it might be better to simply allow this 4 byte string to reside in SRAM and move on.
